### PR TITLE
feat: allow/revoke temp group access

### PR DESCRIPTION
**Pull request by htztomic**
Issue #, if available:

Description of changes:

Checklist:

* [ ] Have you successfully deployed to an AWS account with your changes?
* [ ] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationMiddleware.test.ts [hunk 1]
```diff
@@ -37,7 +37,12 @@ describe('dynamicAuthorizationMiddleware tests', () => {
       removeUserFromGroup: jest.fn(),
       getGroupStatus: jest.fn(),
       setGroupStatus: jest.fn(),
-      doesGroupExist: jest.fn()
+      doesGroupExist: jest.fn(),
+      validateUserGroups: jest.fn().mockImplementation((request) => {
```
Comment (otlci) [inline workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationMiddleware.test.ts:41]: nit: `jest.fn().mockReturnValue({validGroupIds: request.groupIds})`

File: workbench-core/user-management/config/jest.config.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
     }
   },
   "coveragePathIgnorePatterns": ["<rootDir>/src/index.ts"],
-  "coverageReporters": ["json-summary", "json", "lcov"],
+  "coverageReporters": ["json-summary", "json", "lcov", "text"],
```
Comment (otlci) [inline workbench-core/user-management/config/jest.config.json:13]: Unintended change?

File: workbench-core/user-management/src/plugins/cognitoUserManagementPlugin.test.ts [hunk 3]
```diff
@@ -1182,4 +1230,84 @@ describe('CognitoUserManagementPlugin tests', () => {
       await expect(plugin.deleteRole(roles[0])).rejects.toThrow(Error);
     });
   });
+
+  describe('validateUserRoles', () => {
```
Comment (otlci) [inline workbench-core/user-management/src/plugins/cognitoUserManagementPlugin.test.ts:1234]: Should we test for expired access entry? E.g. set TTL to 500 ms and verify that expires behaves as expected.

File: workbench-core/user-management/config/jest.config.json (hunk 2 above)
Comment (htztomic) [reply workbench-core/user-management/config/jest.config.json:13]: this helps with displaying coverage

File: workbench-core/user-management/src/plugins/cognitoUserManagementPlugin.test.ts (hunk 3 above)
Comment (htztomic) [reply workbench-core/user-management/src/plugins/cognitoUserManagementPlugin.test.ts:1234]: it does check for expiration under the test `Do not modify roles if user is temporarily revoked access but has expired`

File: workbench-core/example/infrastructure/integration-tests/tests/dynamicAuthorization/identityPermissions.test.ts [hunk 4]
```diff
@@ -323,7 +323,7 @@ describe('dynamic authorization identity permission integration tests ', () => {
     let subjectId: string;
     let parentId: string;
     let randomTextGenerator: RandomTextGenerator;
-    let newAdminSession: ClientSession;
+    // let newAdminSession: ClientSession;
```
Comment (kpark277) [inline workbench-core/example/infrastructure/integration-tests/tests/dynamicAuthorization/identityPermissions.test.ts:326]: nit: delete if not necessary

File: workbench-core/example/infrastructure/integration-tests/tests/dynamicAuthorization/identityPermissions.test.ts [hunk 5]
```diff
@@ -365,11 +365,11 @@ describe('dynamic authorization identity permission integration tests ', () => {
       await adminSession.resources.groups
         .group(groupId)
         .addUser({ userId: adminSession.getSettings().get('rootUserId') });
-      newAdminSession = await setup.createAdminSession();
+      // newAdminSession = await setup.createAdminSession();
```
Comment (kpark277) [inline workbench-core/example/infrastructure/integration-tests/tests/dynamicAuthorization/identityPermissions.test.ts:368]: Delete if not necessary.

**Review by kpark277 — APPROVED** [review]

**Pull request merged**